uint32 nDimMax
uint32 nDim
singlePath[1] path
  




